### PR TITLE
Task/setup redux observable

### PR DESCRIPTION
**Pull request by smyaseen**
Configured and setted up redux-observable

### CODE REVIEW

File: src/redux/actions/login-logout.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const LOGIN = 'LOGIN';
```
Comment (SiddiquiMonis401) [inline src/redux/actions/login-logout.js:1]: We have to make a separate directory for action types. All the constants will be defined there ..

File: src/redux/store.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { createStore, applyMiddleware } from 'redux';
+import { createEpicMiddleware } from 'redux-observable';
+import { Reducers } from './reducers';
+import { Epics } from './epics';
+
+const epicMiddleware = createEpicMiddleware();
+
+export default createStore(Reducers.loginLogout, applyMiddleware(epicMiddleware));
```
Comment (SiddiquiMonis401) [inline src/redux/store.js:8]: In create store we should use the combine reducer function that is to combine the multiple reducer in a single store..

File: src/redux/reducers/login-logout.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { Actions } from '../actions';
+
+export const loginLogout = (state = { isLoggedIn: false }, action) => {
```
Comment (SiddiquiMonis401) [inline src/redux/reducers/login-logout.js:3]: As there will be a single export from our reducer file so try to use default export from here..

File: src/App.js [hunk 4]
```diff
@@ -1,25 +1,24 @@
...
+        Log In
+      </button>
     </div>
   );
-}
+};
 
-export default App;
+const mapStateToProps = (state) => ({
+  isLoggedIn: state.isLoggedIn,
```
Comment (SiddiquiMonis401) [inline src/App.js:17]: We will be using the (useSelector and useDispatch) hook for functional component. So avoid connect in functional components..

File: src/redux/actions/login-logout.js (hunk 1 above)
Comment (smyaseen) [reply src/redux/actions/login-logout.js:1]: Alright, Working on it.

Comment (smyaseen) [reply src/redux/actions/login-logout.js:1]: Resolved

File: .eslintrc.json [hunk 5]
```diff
@@ -34,13 +34,14 @@
...
         {
+            "endOfLine": "auto",
             "printWidth": 120,
             "singleQuote": true,
             "trailingComma" : "all",
             "arrow-parens": "avoid"
         }
     ],
-    "linebreak-style": ["error", "unix"],
+    "linebreak-style": ["error", "windows"],
```
Comment (umair-khanzada) [inline .eslintrc.json:44]: I won't suggest you change in lint file.. it's a pre define config and we need to follow it.

**Pull request merged**